base-theme/assets/webpack/webpack.dev.ts

```diff
@@ -5,6 +5,11 @@ import { CleanWebpackPlugin } from "clean-webpack-plugin"
 import MiniCssExtractPlugin from "mini-css-extract-plugin"
 import common from "./webpack.common"
 import "webpack-dev-server" // this import tells webpack's typings about the devServer type
+import * as envalid from "envalid"
+
+const env = envalid.cleanEnv(process.env, {
+  WEBPACK_PORT: envalid.port()
+})
```

Comment:
This is https://www.npmjs.com/package/envalid. It's nice for (a) validating that env variables are set and providing good error messages if they are not, and (b) typescript. Since `process.env.WHATEVER` is `string | undefined` by default.

I've scattered a few `envalid` calls into the repo. I *think* one common usage of envalid is to have a single file that validates the environment variables and sets defaults and stuff. (Like `settings.py` in Django, I guess). I think that would be really nice, but we'd have to have some centralized js-based cli through which we call hugo (Rather than our current shell scripts). I briefly explored doing something like that with [oclif](https://oclif.io/), but it seemed like way too big a change for the moment.

*Aside:  [oclif](https://oclif.io/) is cool. And it's built by Salesforce / it's what the heroku CLI is built upon. And while I like the idea of a centralized CLI for this repo, like django provides us elsewhere, I would not recommend `oclif` because https://npmtrends.com/commander-vs-oclif-vs-yargs*